### ISSUE TITLE
IRetryAnalyzer bug in 6.8.17

### ISSUE THREAD

**Issue by daniel-git** [cbeust/testng#604]
In the version 6.8.17 the IRetryAnalyzer does not work correct. In the follow example shows a test that should run 3 times and every test should fail. In version 6.8.13 this works correct. In 6.8.17 and 6.8.21 the test rerun forever (until a java.lang.OutOfMemoryError: Java heap space)

```java
public class RetryTest {

    public static class MyRA implements IRetryAnalyzer {

        int i=0;
        
        @Override
        public boolean retry(ITestResult result) {
            i++;
            boolean retry = i <= 2;
            System.out.println("retry " + retry + " count: " + i);
            return retry;
        }

    }

    @Test(retryAnalyzer = MyRA.class)
    public void hello() {
        System.out.println("hello");
        throw new RuntimeException();
    }
}
```

The Output is:
```
hello
retry true count: 1
hello
retry true count: 2
retry false count: 3
hello
retry false count: 4
retry false count: 5
retry false count: 6
hello
retry false count: 7
retry false count: 8
retry false count: 9
retry false count: 10
...
```

**Comment by rschmitt** [comment]
I can't reproduce this. Are you sure this isn't due to a thread safety issue in the test you wrote? What happens if you make `i` a `final AtomicInteger`?

**Comment by rschmitt** [comment]
Oh nevermind, it looks like this was fixed after the 6.8.21 release by https://github.com/cbeust/testng/commit/6c40ac103c04fac38274dafc077d55df12e9d8d7.
